source/Unittests/TheService.cs

```diff
@@ -99,6 +99,23 @@ public void DoesNotThrowsExceptionForUnknownLocales(string locale)
             var dontCare = service.GetString(IdSrvConstants.Messages, MessageIds.MissingClientId);
         }
 
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void GetLocaleReturnsDefaultWhenLocaleProviderReturnsNullOrWhitespace(string locale)
```

Comment:
This was already covered by `DoesNotThrowsExceptionForUnknownLocales` (except for checking the returned string) - I renamed that test to `UnknownLocalesUseDefaultLocale` and check the returned result.